app/models/katello/ping.rb

```diff
@@ -137,6 +143,25 @@ def pulp_without_auth(url)
         json
       end
 
+      def pulp3_without_auth(url)
+        json = backend_status(url, :pulp)
+        fail _("Pulp does not appear to be running at %s.") % url if json.empty?
+
+        if json['database_connection'] && json['database_connection']['connected'] != true
+          fail _("Pulp database connection issue at %s.") % url
+        end
+
+        if json['redis_connection'] && json['redis_connection']['connected'] != true
+          fail _("Pulp redis connection issue at %s.") % url
+        end
+
+        if json["online_workers"].empty?
+          fail _("Not all necessary pulp workers running at %s.") % url
+        end
```

Comment:
So the status api returns the pulp-resource-manager service as a worker. If we just stop the workers then the tasks end up waiting for workers to be started. I guess we can ask for a separate key for workers other than the resource manager itself or check if the count>1 of workers?